stripe/src/main/java/com/stripe/android/view/CardFormView.kt

```diff
@@ -0,0 +1,450 @@
+package com.stripe.android.view
+
+import android.content.Context
+import android.content.res.ColorStateList
+import android.graphics.Color
+import android.text.Editable
+import android.util.AttributeSet
+import android.util.TypedValue
+import android.view.LayoutInflater
+import android.view.inputmethod.EditorInfo
+import android.widget.LinearLayout
+import androidx.annotation.VisibleForTesting
+import androidx.core.content.ContextCompat
+import androidx.core.content.withStyledAttributes
+import androidx.core.view.isVisible
+import androidx.core.view.updateLayoutParams
+import androidx.core.widget.doAfterTextChanged
+import com.stripe.android.R
+import com.stripe.android.databinding.StripeCardFormViewBinding
+import com.stripe.android.databinding.StripeHorizontalDividerBinding
+import com.stripe.android.databinding.StripeVerticalDividerBinding
+import com.stripe.android.model.Address
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.CardParams
+import com.stripe.android.view.CardFormView.Style
+import com.stripe.android.view.CardValidCallback.Fields
+
+/**
+ * A view to collect credit card information and provide [CardParams] for API invocation.
+ *
+ * Use [R.styleable.StripeCardFormView_cardFormStyle] to toggle style between [Style.Standard] and [Style.Borderless],
+ * Use [R.styleable.StripeCardFormView_formBackgroundColorStateList] to change the card's background color in enable and disabled state.
+ *
+ * To access the [CardParams], see details in [cardParams] property.
+ * To get notified if the current card params are valid, set a [CardValidCallback] object with [setCardValidCallback].
+ */
+class CardFormView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+    private val layoutInflater = LayoutInflater.from(context)
+    private val viewBinding = StripeCardFormViewBinding.inflate(layoutInflater, this)
+
+    private val cardContainer = viewBinding.cardMultilineWidgetContainer
+
+    @VisibleForTesting
+    internal val cardMultilineWidget = viewBinding.cardMultilineWidget
+
+    @VisibleForTesting
+    internal val countryPostalDivider = viewBinding.countryPostalDivider
+
+    @VisibleForTesting
+    internal val postalCodeContainer = viewBinding.postalCodeContainer
+
+    @VisibleForTesting
+    internal val errors = viewBinding.errors
+
+    @VisibleForTesting
+    internal val postalCodeView = viewBinding.postalCode
+
+    @VisibleForTesting
+    internal val countryLayout = viewBinding.countryLayout
+
```

Comment:
Instead of making these views `internal` when they are only needed for testing, you can do something like. I really like this pattern.

```
@Test
fun `when locale is US then country should be US and postal config should be US`() {
    val viewBinding = StripeCardFormViewBinding.bind(standardCardFormView)
    assertThat(viewBinding.postalCode.text).isEqualTo("something")
}
```

Reply:
neat!